src/compiler/moduleNameResolver.ts

```diff
@@ -188,7 +188,10 @@ namespace ts {
 
         let resolvedTypeReferenceDirective: ResolvedTypeReferenceDirective | undefined;
         if (resolved) {
-            resolved = realpath(resolved, host, traceEnabled);
+            if (!options.preserveSymlinks) {
+                resolved = realpath(resolved, host, traceEnabled);

```

Comment:
Can you add a doc-comment describing what `ModuleResolutionHost.realpath` (called by `realpath`) is supposed to do? It's not obvious to me from the name that it is supposed to resolve symbolic links.